calling-apis/managing-asynchronous-calls.md

```diff
@@ -0,0 +1,277 @@
+  })
+  .catch( (error) => {
+    console.log('error in findLocation!');
+  });
+}
+
+const findLocationFromLatAndLon = (query) => {
+  let latitude, longitude;
+  axios.get('https://us1.locationiq.com/v1/search.php',
+  {
+    params: {
+      key: YOUR_API_KEY,
+      q: 'Seattle, Washington, USA',
+      format: 'json'
+    }
+  })
+  .then( (response) => {
+    latitude = response.data[0].lat;
+    longitude = response.data[0].lon;
+    console.log('success in findLatitudeAndLongitude!', latitude, longitude);
+
+    findLocation(latitude, longitude);
+  })
+  .catch( (error) => {
+    console.log('error in findLatitudeAndLongitude!');
+  });
+}
+
+findLocationFromLatAndLon('Seattle, Washington, USA');
+```
+<!-- prettier-ignore-end -->
+
+Read through this code, and then answer the questions below.
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: 72400dc5
+* title: Managing Asynchronous Calls
+##### !question
+
+Which of the following most accurately describes where we invoke `findLocation`?
+
+##### !end-question
+##### !options
+
+* After we define `findLocation`
+* After we invoke `findLatitudeAndLongitude`
+* Inside the `then` block that is chained after making the first API call
+* Inside `findLatitudeAndLongitude`, after making the first API call
+
+##### !end-options
+##### !answer
+
+* Inside the `then` block that is chained after making the first API call
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: 9b3c1438
+* title: Managing Asynchronous Calls
+##### !question
+
+When we invoke `findLocation`, we pass in a latitude and longitude. Where did these values come from?
+
+##### !end-question
+##### !options
+
+* The response from the first API call
+* The response from the second API call
+* The return value of `findLocation`
+* The return value of `findLatitudeAndLongitude`
+
+##### !end-options
+##### !answer
+
+* The response from the first API call
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: fa009186
+* title: Managing Asynchronous Calls
+##### !question
+
+Which of the following most accurately describes where we print the results from the second API call?
+
+##### !end-question
+##### !options
+
+* After we invoke `findLocation`
+* After we invoke `findLatitudeAndLongitude`
+* Inside the `then` block that is chained after making the second API call
+* Inside `findLocation`, after making the first API call
+
+##### !end-options
+##### !answer
+
+* Inside the `then` block that is chained after making the second API call
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+When we introduce other tools, libraries, and programming paradigms, we will approach this from a different angle.
```

Comment:
rather than saying we'll explore some other approaches later, let's start the discussion about where the limitations of this solution are, and give similar pointers as in other lessons about where the reader might continue their own investigation.
```suggestion
This solution successfully uses the results of one API call to make another API call, but it has a few drawbacks:

- `findLatitudeAndLongitude` _does_ find the coordinates, but that's not _all_ it does. At the least, we might want to rename the function to more accurately describe what the entire chain of functionality does. In general, we might say this method now violates the Single Responsibility Principle.
- What if we _do_ only need to get the coordinates? By calling the second method directly from inside the first, we have linked these two functions together, so it's harder to use one without the other.
- What if there were a third operation we needed to perform using the location result that was looked up by latitude and longitude? Would we call that third method from inside the second? Would that make it difficult to keep track of what this growing chain of function calls is doing?

### !callout-info

## Promises, Promises

There are many ways to approach situations like this. For some ideas on where to start our own research, we could look for information about "Promise chaining" or "JavaScript async await". We could also think about how we could use callback functions to help us customize the behavior here. Follow your curiosity!

### !end-callout
```